tests/com/magento/idea/magento2plugin/inspections/xml/PreferenceDeclarationInspectionTest.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+package com.magento.idea.magento2plugin.inspections.xml;
+
+import com.magento.idea.magento2plugin.magento.files.ModuleDiXml;
+
+public class PreferenceDeclarationInspectionTest extends InspectionXmlFixtureTestCase {
+
+    private static final String ARGUMENT_VALUE_IS_EMPTY =
+            "inspection.error.idAttributeCanNotBeEmpty";
+
+    private static final String CLASS_DOES_NOT_EXIST =
+            "inspection.warning.class.does.not.exist";
+
+    private static final String EXISTENT_CLASS_ONE =
+            "Magento\\Catalog\\Api\\ProductRepositoryInterface";
+
+    private static final String EXISTENT_CLASS_TWO =
+            "Foo\\Bar\\Model\\Logger";
+
+    private static final String NOT_EXISTENT_CLASS =
+            "Not\\Existent\\Class";
+
+    @Override
+    public void setUp() throws Exception {
+        super.setUp();
+        myFixture.enableInspections(PreferenceDeclarationInspection.class);
+    }
+
+    /**
+     * Test for an error for the "type" attribute because it is empty.
+     * <preference for="" type="Foo\Bar\Model\Logger"/>
+     */
+    public void testAttrArgForValuesIsEmpty() {
+        configureFixture();
+
+        final String forAttrIsEmptyMessage =  inspectionBundle.message(
+                ARGUMENT_VALUE_IS_EMPTY,
+                ModuleDiXml.TYPE_ATTR
+        );
+
+        assertHasHighlighting(forAttrIsEmptyMessage);
+    }
+
+    /**
+     * Test for an error for the "for" attribute because it is empty.
+     * <preference for="Foo\Bar\Model\Logger" type=""/>
+     */
+    public void testAttrArgTypeValueIsEmpty() {
```

Comment:
@Iamwade, there is the same confusing...you are testing for attribute, but the method, test data directory and XML in the description opposite. Please, check it.

Reply:
Fixed.

Reply:
Thank you!